services:
  pygeoapi:
    build:
      context: .
      dockerfile: pygeoapi/Dockerfile
    ports:
      - "5000:80"
    volumes:
      - ./pygeoapi-config.yml:/pygeoapi/local.config.yml
      - ./catalogue.tinydb:/pygeoapi/catalogue.tinydb
    networks:
      - pygeoapinet
    depends_on:
      postgres:
        condition: service_healthy

  postgres:
    image: postgis/postgis
    restart: always
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"]
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    networks:
      - pygeoapinet
    volumes:
      - postgres:/var/lib/postgresql/data
    ports:
      - "5432:5432"

volumes:
  postgres:

networks:
  pygeoapinet:
    name: pygeoapi_network
